IonHashDotNet/IonHashWriter.cs

```diff
@@ -0,0 +1,342 @@
+        {
+            get;
+            private set;
+        }
+
+        public string FieldName
+        {
+            get;
+            private set;
+        }
+
+        public bool IsNull
+        {
+            get;
+            private set;
+        }
+
+        public IonType Type
+        {
+            get;
+            private set;
+        }
+
+        public dynamic Value
+        {
+            get;
+            private set;
+        }
+
+        // implements IIonWriter
+        public ISymbolTable SymbolTable
+        {
+            get { return this.writer.SymbolTable; }
+        }
+
+        public bool IsInStruct
+        {
+            get { return this.writer.IsInStruct; }
+        }
+
+        // implements IIonHashWriter
+        public byte[] Digest()
+        {
+            return this.hasher.Digest();
+        }
+
+        public void AddTypeAnnotation(string annotation)
+        {
+            this.AddTypeAnnotationSymbol(new SymbolToken(annotation, SymbolToken.UnknownSid));
+            this.writer.AddTypeAnnotation(annotation);
+        }
+
+        public void AddTypeAnnotationSymbol(SymbolToken annotation)
+        {
+            this.Annotations.Add(annotation);
+            this.writer.AddTypeAnnotationSymbol(annotation);
+        }
+
+        public void ClearTypeAnnotations()
+        {
+            this.Annotations.Clear();
+            this.writer.ClearTypeAnnotations();
+        }
+
+        public void Dispose()
+        {
+            this.writer.Dispose();
+        }
+
+        public void Finish()
+        {
+            this.writer.Finish();
+        }
+
+        public void Flush()
+        {
+            this.writer.Flush();
+        }
+
+        public void SetFieldName(string name)
+        {
+            this.FieldName = name;
+            this.writer.SetFieldName(name);
+        }
+
+        public void SetFieldNameSymbol(SymbolToken symbol)
+        {
+            this.FieldName = symbol.Text;
+            this.writer.SetFieldNameSymbol(symbol);
+        }
+
+        public void SetTypeAnnotations(IEnumerable<string> annots)
+        {
+            this.Annotations.Clear();
+            foreach (var annotation in annots)
+            {
+                this.AddTypeAnnotationSymbol(new SymbolToken(annotation, SymbolToken.UnknownSid));
+            }
+
+            this.writer.SetTypeAnnotations(annots);
+        }
+
+        public void StepIn(IonType type)
+        {
+            this.Type = type;
+            this.Value = null;
+            this.IsNull = false;
+            this.hasher.StepIn(this);
+            this.writer.StepIn(type);
+            this.FieldName = default;
+            this.Annotations.Clear();
+        }
+
+        public void StepOut()
+        {
+            this.hasher.StepOut();
+            this.writer.StepOut();
+        }
+
+        public void WriteBlob(ReadOnlySpan<byte> value)
+        {
+            List<byte> byteList = new List<byte>(value.ToArray());
```

Comment:
not sure i understand the reason for creating a List?

Reply:
It is because the IonWriter API takes in a ReadOnlySpan<byte>, but that type cannot be assigned to generic functions of "dynamic" type so internally we'll use a List.

Reply:
is `byte[]` an option?

Reply:
Yes, it seems to work. Changed!